lib/persistence/dao/dao.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:battle_it_out/persistence/entities/dto.dart';
+import 'package:battle_it_out/persistence/database_provider.dart';
+import 'package:sqflite/sqflite.dart';
+
+abstract class DAO<T extends DTO> {
+  get tableName;
+
+  dynamic fromMap(Map<String, dynamic> map);
+
+  Future<T> get(int id) async {
```

Comment:
zmieniłbym nazwę tego na jakieś read albo retrieve. Nie wiem jak ty, ale jak jest get, to od razu myślę, że jest też set, i pewnie put, a nie create i update. No chyba że chcesz zmienić pozostałe funkcje na set i put :D

Reply:
zmieniam nazewnictwo na get, set, put, delete